cmd/zpool/zpool_main.c

```diff
@@ -1107,6 +1107,11 @@ zpool_do_create(int argc, char **argv)
 		}
 	}
 
+	/* Make new pools default to partition=raw */
+	if (add_prop_list_default(zpool_prop_to_name(ZPOOL_PROP_PARTITION),
+	    "raw", &props, B_TRUE))
+		goto errout;
```

Comment:
> So I have a few questions, would using raw device cause it to not align to 1MB?

Unless ZFS creates -part1 which is explicitly aligned at 1MB things will just be aligned at the start of the device.  That's usually fine for standard hard drives but it can lead to performance problems with other kinds block devices which may include a small header which can mess up the alignment.  This has lead to issues with Linux multipath devices or harware/software RAID device which have an internal RAID stripe size.  Starting at a 1MB offset in to the device helps avoid these kind of issues.

> And the leaving space at the end is only done current on wholedisk=1 case?

Right.  The `wholedisk=0` case is effectively the same as the raw case.